Source/WebCore/bindings/js/JSObservableArray.cpp

```diff
@@ -122,8 +122,11 @@ bool JSObservableArray::put(JSCell* cell, JSGlobalObject* lexicalGlobalObject, P
     auto scope = DECLARE_THROW_SCOPE(vm);
 
     JSObservableArray* thisObject = jsCast<JSObservableArray*>(cell);
-    if (propertyName == vm.propertyNames->length)
-        return false;
+    if (propertyName == vm.propertyNames->length) {
+        auto length = value.toUInt32(lexicalGlobalObject);
+        RETURN_IF_EXCEPTION(scope, false);
+        return thisObject->getConcreteArray().setLength(length);
+    }
```

Comment:
I think `::defineOwnProperty` implementation and `deleteProperty`'s length support is necessary, so let's add them.
Can you also include tests for them?